### PR TITLE
Restructure looks of compare.html site

### LINKED ISSUE #776: Narrowing the layout of compare site
The compare site has the submit button really far away from two commit text-boxes.
Can we make the submit button near the stat button ?
This way the mouse cursor doesn't have to move far away to reach submit button.

image

Also, can we narrow down the width of layout ? About 800px ? We have a lot of unused blank spaces
in desktop screen.
image

### PR DESCRIPTION
**Pull request by lzutao**
This PR:
* adjusted the look of input form in compare.html site.
* limited body max-width at 800px:
    * there are much blank space in desktop.
    * to move content to screen central.
* added lame support (just a meta header) for mobile.

Closes #776.

---

Below are some comparison of old and new looks.

## Old

<details><summary>Top</summary>

image
</details>

<details><summary>Bottom</summary>

image
</details>

## New

<details><summary>Top</summary>

image
</details>

<details><summary>Bottom</summary>

image
</details>

On emulated phone mode:

<details><summary>Warning: big picture</summary>

image
</details>

### CODE REVIEW

File: site/static/compare.html [hunk 1]
```diff
@@ -46,18 +47,20 @@ <h3>Cache states</h3>
...
-            <select id='stats' name="stat">
-            </select>
-            <div class="submit">
-                <a href="#" onClick="submit_settings(); return false;">Submit</a>
-            </div>
-        </div>
-    </div>
+    <form id="settings" action="">
+        <fieldset id="commits">
+            <legend>Commits</legend>
```
Comment (Julian-Wollersberger) [inline site/static/compare.html:52]: Maybe you could adjust the legend to eg "Select comparision" ~~or "Choose commits to compare"~~ ?

File: site/static/compare.html [hunk 2]
```diff
@@ -46,18 +47,20 @@ <h3>Cache states</h3>
...
+            <label for="start-bound">Commit/Date A:</label>
+            <input width="100em" placeholder="YYYY-MM-DD or SHA" id="start-bound" /><br>
+            <label for="end-bound">Commit/Date B:</label>
+            <input width="100em" placeholder="YYYY-MM-DD or SHA" id="end-bound" /><br>
+        </fieldset>
+        <label for="stats">Choose a comparison method:</label>
+        <select id='stats' name="stat">
+        </select><br>
+        <input type="submit" value="Submit" onclick="submit_settings(); return false;">
```
Comment (Julian-Wollersberger) [inline site/static/compare.html:61]: I'd include the buttons in the fieldset so the box is drawn around all of it.

**Review by Julian-Wollersberger — COMMENTED** [review]
This definitely improves the readability.
Especially the submit button, which was easy to overlook previously.

File: site/static/compare.html (hunk 2 above)
Comment (lzutao) [reply site/static/compare.html:61]: I don't have strong preference. To be honest, I'm not a web dev. This PR is result of googling around
plus some web basic knowledge. I'd leave it to maintainer to decide.

File: site/static/compare.html [hunk 3]
```diff
@@ -2,6 +2,7 @@
 <html>
 <head>
 <meta charset="utf-8">
+<meta name="viewport" content="width=device-width, initial-scale=1.0">
```
Comment (Mark-Simulacrum) [inline site/static/compare.html:5]: It always worked fairly okay for me already on mobile, what sort of improvement does this bring?

Comment (lzutao) [reply site/static/compare.html:5]: This is current page on mobile, we have to zoon in to view the page.
<details><summary>Old change</summary>

Screen Shot 2020-10-09 at 09 40 55
</details>

With this meta tag, we don't. MDN wrote something about this: https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Responsive_Design#The_viewport_meta_tag

**Comment by varkor** [comment]
I personally frequently found myself looking for the submit button in the wrong place, so I'm glad to see this!

**Pull request merged**